src/v/kafka/client/utils.h

```diff
@@ -56,8 +56,8 @@ auto retry_with_mitigation(
                           // ignore failed mitigation
                       });
                 }
-                return fut.then(func).handle_exception(
-                  [&eptr](std::exception_ptr ex) mutable {
+                return fut.then([&func]() mutable { return func(); })
```

Comment:
you can probably drop `mutable` here since the capture is a reference which isn't changing even if the thing it is referring to is changing.